pkg/hyperkit/driver.go

```diff
@@ -169,137 +244,247 @@ func (d *Driver) Remove() error {
 	return nil
 }
 
+// Restart a host
 func (d *Driver) Restart() error {
 	return pkgdrivers.Restart(d)
 }
 
 // Start a host
 func (d *Driver) Start() error {
-	h, err := hyperkit.New("", "", filepath.Join(d.StorePath, "machines", d.MachineName))
-	if err != nil {
+	if err := d.verifyRootPermissions(); err != nil {
 		return err
 	}
 
+	stateDir := filepath.Join(d.StorePath, "machines", d.MachineName)
+	if err := d.recoverFromUncleanShutdown(); err != nil {
+		return err
+	}
+	h, err := hyperkit.New("", d.VpnKitSock, stateDir)
+	if err != nil {
+		return errors.Wrap(err, "new-ing Hyperkit")
+	}
+
 	// TODO: handle the rest of our settings.
-	h.Kernel = d.ResolveStorePath(d.Vmlinuz)
-	h.Initrd =d.ResolveStorePath(d.Initrd)
+	h.Kernel = d.BootKernel
+	h.Initrd = d.BootInitrd
 	h.VMNet = true
 	h.ISOImages = []string{d.ResolveStorePath(isoFilename)}
 	h.Console = hyperkit.ConsoleFile
-	h.CPUs = d.CPU
-	h.Memory = d.Memory
+	if d.CPU > defaultCPUs {
+		h.CPUs = d.CPU
+	}
+	if d.Memory > defaultMemory {
```

Comment:
```suggestion
	if d.Memory != 0 {
```
While unlikely and against better judgement, user should be allowed to lower the request for memory. Same could be done for CPU count, if fractions of CPU are supported by hyperkit.